### PR TITLE
[CARBONDATA-3772] Update index documents

### PR DESCRIPTION
**Pull request by ShreelekhyaG**
### Why is this PR needed?
 update index documentation to comply with recent changes
 
 ### What changes were proposed in this PR?

    
 ### Does this PR introduce any user interface change?
 - No
 - Yes. (please explain the change and update document)

 ### Is any new testcase added?
 - No
 - Yes

### CODE REVIEW

**Comment by ajantha-bhat** [comment]
Add to whitelist

**Comment by ajantha-bhat** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1026/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2739/

File: docs/index/bloomfilter-index-guide.md [hunk 1]
```diff
@@ -109,40 +110,40 @@ User can create BloomFilter DataMap using the Create DataMap DDL:
 ## Loading Data
 When loading data to main table, BloomFilter files will be generated for all the
 index_columns given in DMProperties which contains the blockletId and a BloomFilter for each index column.
-These index files will be written inside a folder named with DataMap name
+These index files will be written inside a folder named with Index name
```
Comment (Indhumathi27) [inline docs/index/bloomfilter-index-guide.md:113]: Please change the statement in Line no.112. `When loading data to main table, BloomFilter files will be generated for all the index_columns provided in the CREATE statement which contains the blockletId`

File: docs/index/bloomfilter-index-guide.md [hunk 2]
```diff
@@ -109,40 +110,40 @@ User can create BloomFilter DataMap using the Create DataMap DDL:
...
+User can verify whether a query can leverage BloomFilter Index by executing `EXPLAIN` command,
+which will show the transformed logical plan, and thus user can check whether the BloomFilter Index can skip blocklets during the scan.
+If the Index does not prune blocklets well, you can try to increase the value of property `BLOOM_SIZE` and decrease the value of property `BLOOM_FPP`.
 
-## Data Management With BloomFilter DataMap
-Data management with BloomFilter DataMap has no difference with that on Lucene DataMap.
-You can refer to the corresponding section in `CarbonData Lucene DataMap`.
+## Data Management With BloomFilter Index
+Data management with BloomFilter Index has no difference with that on Lucene Index.
+You can refer to the corresponding section in `CarbonData Lucene Index`.
```
Comment (Indhumathi27) [inline docs/index/bloomfilter-index-guide.md:125]: Looks like `CarbonData Lucene Index` link is not working. Please fix

File: docs/index/index-management.md [hunk 3]
```diff
@@ -18,51 +18,49 @@
...
 ```
 
-Currently, there are 5 DataMap implementations in CarbonData.
+Currently, there are 3 Index implementations in CarbonData.
 
-| DataMap Provider | Description                              | DMPROPERTIES                             | Management       |
+| Index Provider | Description                              | PROPERTIES                             | Management       |
 | ---------------- | ---------------------------------------- | ---------------------------------------- | ---------------- |
-| mv               | multi-table pre-aggregate table          | No DMPROPERTY is required                | Manual/Automatic           |
+| secondary-index  | secondary-index tables to hold blocklets as indexes and managed as child tables        | No PROPERTY is required                | Automatic           |
```
Comment (Indhumathi27) [inline docs/index/index-management.md:47]: Can remove Properties column,  as now, index_columns will be provided in CREATE statement and not required to give in properties

File: docs/index/index-management.md [hunk 4]
```diff
@@ -73,69 +71,69 @@ If user perform following command on the main table, system will return failure.
     not, the operation is allowed, otherwise operation will be rejected by throwing exception.
 3. Partition management command: `ALTER TABLE ADD/DROP PARTITION`.
 
-If user do want to perform above operations on the main table, user can first drop the datamap, perform the operation, and re-create the datamap again.
+If user do want to perform above operations on the main table, user can first drop the index, perform the operation, and re-create the index again.
```
Comment (Indhumathi27) [inline docs/index/index-management.md:74]: Change from `pre-aggregate table` to `index` in Line no.70

File: docs/index/index-management.md [hunk 5]
```diff
@@ -73,69 +71,69 @@ If user perform following command on the main table, system will return failure.
...
  By default it is automatic refresh,
- which means its data will get refreshed immediately after the datamap is created or the main table is loaded.
- Manual refresh on this datamap will has no impact.
+ which means its data will get refreshed immediately after the index is created or the main table is loaded.
+ Manual refresh on this index will has no impact.
 
 
 
-## DataMap Catalog
+## Index Catalog
```
Comment (Indhumathi27) [inline docs/index/index-management.md:98]: Please remove this section, as all indexes will be stored in carbontable itself in tableproperties

File: docs/index/index-management.md [hunk 6]
```diff
@@ -73,69 +71,69 @@ If user perform following command on the main table, system will return failure.
...
++- pruning by CG Index
 +- all blocklets: 1
    skipped blocklets: 0
 ```
 
-### Show DataMap
+### Show Index
 
-There is a SHOW DATAMAPS command, when this is issued, system will read all datamap from *system* folder and print all information on screen. The current information includes:
+There is a SHOW INDEXES command, when this is issued, system will read all index from *system* folder and print all information on screen. The current information includes:
```
Comment (Indhumathi27) [inline docs/index/index-management.md:131]:
```suggestion
There is a SHOW INDEXES command, when this is issued, system will read all index from the carbon table and print all information on screen. The current information includes:
```

File: docs/index/index-management.md [hunk 7]
```diff
@@ -73,69 +71,69 @@ If user perform following command on the main table, system will return failure.
...
 
-- DataMapName
-- DataMapProviderName like mv
+- IndexName
```
Comment (Indhumathi27) [inline docs/index/index-management.md:133]: Please check and update show index information

File: docs/index/lucene-index-guide.md [hunk 8]
```diff
@@ -15,75 +15,74 @@
...
    cardinality dimensions.
-3. SPLIT_BLOCKLET: when made as true then store the data in blocklet wise in lucene , it means new 
+2. SPLIT_BLOCKLET: when made as true then store the data in blocklet wise in lucene , it means new 
    folder will be created for each blocklet, thus, it eliminates storing blockletid in lucene and 
    also it makes lucene small chunks of data.
    
 ## Loading data
 When loading data to main table, lucene index files will be generated for all the
 index_columns(String Columns) given in DMProperties which contains information about the data
-location of index_columns. These index files will be written inside a folder named with datamap name
```
Comment (Indhumathi27) [inline docs/index/lucene-index-guide.md:86]:
```
When loading data to main table, lucene index files will be generated for all the
index_columns(String Columns) given in CREATE statement which contains information about the data
```

**Comment by Indhumathi27** [comment]
@ShreelekhyaG Please check and update links in` README.md` file

File: docs/index/bloomfilter-index-guide.md (hunk 1 above)
Comment (ShreelekhyaG) [reply docs/index/bloomfilter-index-guide.md:113]: Ok

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1034/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2747/

File: README.md [hunk 9]
```diff
@@ -55,9 +55,9 @@ CarbonData is built using Apache Maven, to [build CarbonData](https://github.com
  * [Configuring CarbonData](https://github.com/apache/carbondata/blob/master/docs/configuration-parameters.md) 
  * [DataMap Developer Guide](https://github.com/apache/carbondata/blob/master/docs/datamap-developer-guide.md) 
  * [Data Types](https://github.com/apache/carbondata/blob/master/docs/supported-data-types-in-carbondata.md) 
-* [CarbonData DataMap Management](https://github.com/apache/carbondata/blob/master/docs/datamap/datamap-management.md) 
- * [CarbonData BloomFilter DataMap](https://github.com/apache/carbondata/blob/master/docs/datamap/bloomfilter-datamap-guide.md) 
- * [CarbonData Lucene DataMap](https://github.com/apache/carbondata/blob/master/docs/datamap/lucene-datamap-guide.md) 
+* [CarbonData DataMap Management](https://github.com/apache/carbondata/blob/master/docs/index/index-management.md) 
+ * [CarbonData BloomFilter DataMap](https://github.com/apache/carbondata/blob/master/docs/index/bloomfilter-index-guide.md)
```
Comment (Indhumathi27) [inline README.md:59]: Please change from datamap to index

Comment (ShreelekhyaG) [reply README.md:59]: ok

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2749/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1036/

**Comment by kunal642** [comment]
@ShreelekhyaG Please create a JIRA and update the Title

**Comment by ShreelekhyaG** [comment]
> @ShreelekhyaG Please create a JIRA and update the Title

@kunal642 JIRA created, please review and merge.

**Comment by Indhumathi27** [comment]
LGTM

**Comment by kunal642** [comment]
@ShreelekhyaG Please rebase

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1076/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2788/

**Comment by Indhumathi27** [comment]
@ShreelekhyaG Please update index-developer-guide.md also

File: docs/index-developer-guide.md [hunk 10]
```diff
@@ -18,17 +18,18 @@
...
 
 ### Index Provider
-When user issues `CREATE INDEX index_name ON TABLE main AS 'provider'`, the corresponding DataMapProvider implementation will be created and initialized. 
+When user issues `CREATE INDEX index_name ON TABLE main AS 'provider'`, the corresponding IndexProvider implementation will be created and initialized. 
 Currently, the provider string can be:
-1. class name IndexDataMapFactory implementation: Developer can implement new type of IndexDataMap by extending IndexDataMapFactory
+1. class name IndexFactory implementation: Developer can implement new type of Index by extending IndexFactory
 
-When user issues `DROP INDEX index_name ON TABLE main`, the corresponding DataMapProvider interface will be called.
+When user issues `DROP INDEX index_name ON TABLE main`, the corresponding IndexProvider interface will be called.
```
Comment (Indhumathi27) [inline docs/index-developer-guide.md:32]: Change to `IndexFactory class` instead of `IndexProvider interface`

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2796/

**Comment by Indhumathi27** [comment]
retest this please

**Comment by Indhumathi27** [comment]
LGTM

**Comment by kunal642** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2799/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1087/

**Comment by kunal642** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1092/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2804/

**Comment by kunal642** [comment]
LGTM

**Pull request closed without merging**